### PR TITLE
Oof

### LINKED ISSUE #110: Rewrite client with oof
Hopefully this won't be as big of an ordeal as rewriting it in Riot (#77) - I suppose it depends on how implementing things with [oof](https://github.com/heyitsmeuralex/oof) will be.

I'm thinking we could just add oof as a submodule, underneath the `site` folder? Otherwise we have to figure out a jillion other things like bower and stuff which all won't work with the whole "no building" thing (which is basically the point of oof).

@heyitsmeuralex Obviously we need to design oof, too. I'm thinking avoid dealing with styling - don't bother with scoped CSS (and whatever scary things it might imply, such as shadow DOM and browser incompatibilities). That way we can just reuse ~~[the old CSS](https://github.com/towerofnix/bantisocial/blob/551f15fae247022ba5ec4a012ae37697c94318eb/site/styles.css)~~ [PullJosh's CSS](https://codepen.io/PullJosh/pen/OOYZWK). No need to rewrite *everything!*

### PR DESCRIPTION
**Pull request by towerofnix**
Fixes #110 (or at least begins it).

Demo screenshot

Yeow! This is one big change. Don't feel obligated to review it super indepth-ly.

This gets rid of Riot and Travis (so it closes #111), gets us a *lot* closer to [PullJosh's design](https://codepen.io/PullJosh/pen/OOYZWK), and simplifies the client's code a bunch. It's both smaller in volume (only 300 lines, not including websocket.js and oof, and only one JavaScript file) and in complexity (the code is relatively much simpler than working with Riot, and somewhat even simpler than the old actor model (#5)). Also, error stacks work; there's no compilation voodoo of *any* kind going on here.

While this has more features than the Riot client, it's not a complete implementation of the client (before either of those). Still to be done - but in separate PRs (and in no particular order):

- *Message* `formatting!`
- Save and load data from local storage.
- Edit and delete messages.
- Handle channel rename/delete events.
- Improve date formatting. Showing the full timestamp isn't particularly friendly.
- Prompt/error modals.
- Implement the user list. (This is tracked in #68.)
- Implement avatars. (This is tracked in #101).
- Fix the order channels show up in. (This is tracked in #14.)

NYAN

### CODE REVIEW

**Comment by heyitsmeuralex** [comment]
General points:
* cool and good
* Could do with some more comments
  - especially for 'sections' of the code, e.g. `////////// CHANNEL LIST //////////`

File: site/index.js [hunk 1]
```diff
@@ -0,0 +1,352 @@
+'use strict'
+
+// Utility functions:
+
+function post(path, dataObj) {
+  if (!serverURL.value) {
+    return {error: 'client error - server URL not specified yet'}
```
Comment (heyitsmeuralex) [inline site/index.js]: Surely this should be `return Promise.reject({...})`?

File: site/index.js [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(dataObj)
+  }).then(res => res.json())
+}
+
+function get(path, query = {}) {
+  if (!serverURL.value) {
+    return {error: 'client error - server URL not specified yet'}
```
Comment (heyitsmeuralex) [inline site/index.js]: Same here

File: site/index.js [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+        appendMessage(data.message)
+      }
+
+      if (evt === 'created new channel' && data && data.channel) {
+        sidebarChannelList.append(data.channel)
+      }
+    }
+  })
+
+  serverDict.socket = socket
```
Comment (heyitsmeuralex) [inline site/index.js]: This should be `serverDict[serverHostname].set('socket', socket)` or something. We want to have a separate socket connection kept open _for each server at the same time_.

File: site/index.js [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+
+  activeServerHostname.set(serverHostname)
+}
+
+document.getElementById('add-server').addEventListener('click', () => {
+  const host = prompt('Host URL?')
```
Comment (heyitsmeuralex) [inline site/index.js]: oof no modals

File: site/index.js (hunk 1 above)
Comment (towerofnix) [reply site/index.js]: Probably. It should be noted that server errors also return data ala `{error: 'asdf'}`. Maybe we could make `fetchHelper` automatically throw responses which include an `error` field?

File: site/index.js (hunk 3 above)
Comment (towerofnix) [reply site/index.js]: :+1:, that was a typo!

File: site/index.js (hunk 4 above)
Comment (towerofnix) [reply site/index.js]: Nope, not yet.

File: site/index.js (hunk 1 above)
Comment (towerofnix) [reply site/index.js]: Fixed. Errors are now a lot more error-y. This also simplified the code a fair amount!

**Comment by towerofnix** [comment]
>Could do with some more comments, especially for 'sections' of the code

Good point! Thanks. This should've been obvious but 01:00 me didn't think of it. I've reorganized the code, putting similar things closer to each other, and everything in neat commented sections.

**Pull request merged**